src/go/k8s/pkg/resources/certmanager/type_helpers.go

```diff
@@ -307,13 +312,17 @@ func (cc *ClusterCertificates) prepareAPI(
 		result.nodeCertificate = nodeCert
 	}
 
-	anyListenerWithMutualTLS := false
+	generateClientCerts := false
 	for _, l := range tlsListeners {
-		if l.GetTLS().RequireClientAuth {
-			anyListenerWithMutualTLS = true
+		tls := l.GetTLS()
+		// Trigger client cert generation only if RequireClientAuth is true and
+		// ClientCACertRef hasn't been set.
+		if tls.RequireClientAuth && tls.ClientCACertRef == nil {
```

Comment:
I think it should instead be - if ANY listener has CaCertRef, not generate. Right now it's if any does not have it, generate... maybe that could actually never happen? 🤔